File: JavaScript/01.JavaScript-Basic-February-2021/10.whileLoop-Exersice/05.coins.js
function coin(input) {
    let moneyInlv = Number(input[0]);
    let moneyInCoins = Math.floor(moneyInlv * 100);

    let counter = 0;
    while (moneyInCoins > 0) {
        if (moneyInCoins >= 200) {
            moneyInCoins -= 200;
            counter++;
        } else if (moneyInCoins >= 100) {
            moneyInCoins -= 100;
            counter++;
        } else if (moneyInCoins >= 50) {
            moneyInCoins -= 50;
            counter++;
        } else if (moneyInCoins >= 20) {
            moneyInCoins -= 20;
            counter++
        } else if (moneyInCoins >= 10) {
            moneyInCoins -= 10;
            counter++;
        } else if (moneyInCoins >= 5) {
            moneyInCoins -= 5;
            counter++;
        } else if (moneyInCoins >= 2) {
            moneyInCoins -= 2;
            counter++;
        } else if (moneyInCoins >= 1) {
            moneyInCoins -= 1;
            counter++;
        }
    }
    console.log(counter);
}

coin(["1.23"]);
// coin(["2"]);
// coin(["0.56"]);
// coin(["2.73"]);

File: Python/01.Python-Basic-April-2019/09.while_loop_exercise/05.coins.py
import math

money = float(input())
money *= 100
money=math.floor(money)
counter = 0

while money > 0:
    if money / 200 >= 1:
        counter += 1
        money -= 200
    else:
        if money / 100 >= 1:
            counter += 1
            money -= 100
        else:
            if money / 50 >= 1:
                counter += 1
                money -= 50
            else:
                if money / 20 >= 1:
                    counter += 1
                    money -= 20
                else:
                    if money / 10 >= 1:
                        counter += 1
                        money -= 10
                    else:
                        if money / 5 >= 1:
                            counter += 1
                            money -= 5
                        else:
                            if money / 2 >= 1:
                                counter += 1
                                money -= 2
                            else:
                                if money / 1 >= 1:
                                    counter += 1
                                    money -= 1
print(counter)
